src/simulations/explorer.py

```diff
@@ -19,7 +19,8 @@
 from src.simulations.data_manager import DataManager
 from src.simulations.learning import LearningManager
 from src.simulations.select_actions import ActionSelector
-from src.utilities.userdeftools import (initialise_dict,
+from src.utilities.userdeftools import (_calculate_reactive_power,
```

Comment:
note that functions/methods starting with `_` indicate that they are _private_ methods, i.e. we only access them from the class where they are defined. Here this is not the case so I would rename as `calculate_reactive_power`